File: app/src/main/java/com/nawin/androidmvparchitecture/MvpComponent.java
package com.nawin.androidmvparchitecture;

import android.content.Context;

import com.google.gson.Gson;
import com.nawin.androidmvparchitecture.data.Data;
import com.nawin.androidmvparchitecture.data.remote.DataModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by nawin on 7/20/17.
 */
@Component(modules = {MvpModule.class, DataModule.class})
@Singleton
public interface MvpComponent {
    Context context();

    BuildTypeComponent buildType();

    Data data();

    Gson gson();

    void inject(BaseActivity baseActivity);
}

File: app/src/main/java/com/nawin/androidmvparchitecture/MvpComponent.kt
package com.nawin.androidmvparchitecture

import android.content.Context
import com.nawin.androidmvparchitecture.data.Data
import com.nawin.androidmvparchitecture.data.remote.DataModule
import dagger.Component
import javax.inject.Singleton

/**
 * Created on 11/29/17.
 */
@Component(modules = arrayOf(MvpModule::class, DataModule::class))
@Singleton
interface MvpComponent {
    fun context(): Context

    fun buildType(): BuildTypeComponent

    fun data(): Data

    fun inject(baseActivity: BaseActivity)
}